src/enigma2/Admin.cpp

```diff
@@ -190,9 +190,25 @@ PVR_ERROR Admin::GetDriveSpace(long long *iTotal, long long *iUsed, std::vector<
   {
     std::string capacity;
     std::string freeSpace;
+    std::string mount;
 
     XMLUtils::GetString(hddNode, "e2capacity", capacity);
     XMLUtils::GetString(hddNode, "e2free", freeSpace);
+    XMLUtils::GetString(hddNode, "e2mount", mount);
+
+    if (!mount.empty())
+    {
+      bool validRecordingMount = false;
+
+      for (std::string& location : locations)
+      {
+        if (location.find(mount) != std::string::npos) 
+          validRecordingMount = true;
+      }
+
+      if (!validRecordingMount)
+        continue;
+    }
```

Comment:
good use case for `std::find_if`.

```c++
if (!mount.empty())
{
  auto it = std::find_if(locations.begin(), locations.end(), [&mount](std::string& location) { return location.find(mount) != std::string::npos; });

  if (it == locations.end())
     continue; // no valid mount point
}
```

Even eliminates the need for the bool var.

If you want to stick with your version, please `break;`after `validRecordingMount = true;` No need to iterate further.

Reply:
I like this, but I also like to understand my code ;)

I can't find a good reference to understand the predicate piece: 
`[&mount](std::string& location) { return location.find(mount) != std::string::npos; }`

It's mainly the `[&mount](std::string& location) ` which is not intuitive to me.

Reply:
Ah ha, lambda's. 

[<var(s) I want to pass in>](<var(s) that changes per loop>) { <your condition> }

Reply:
Yes, lambdas. 😉